storm-core/src/jvm/org/apache/storm/zookeeper/Zookeeper.java

```diff
@@ -0,0 +1,355 @@
+        return ret;
+    }
+
+    public static String createNode(CuratorFramework zk, String path, byte[] data, List<ACL> acls) throws RuntimeException {
+        return createNode(zk, path, data, org.apache.zookeeper.CreateMode.PERSISTENT, acls);
+    }
+
+    public static boolean existsNode(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+        Stat stat = null;
+        try {
+            if (watch) {
+                stat = zk.checkExists().watched().forPath(Utils.normalizePath(path));
+            } else {
+                stat = zk.checkExists().forPath(Utils.normalizePath(path));
+            }
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+
+        return stat != null;
+    }
+
+    public static void deleteNode(CuratorFramework zk, String path) throws RuntimeException {
+        try {
+            String npath = Utils.normalizePath(path);
+            if (existsNode(zk, npath, false)) {
+                zk.delete().deletingChildrenIfNeeded().forPath(Utils.normalizePath(path));
+            }
+
+        } catch (KeeperException.NoNodeException e) {
+            LOG.info("exception", e);
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    public static void mkdirs(CuratorFramework zk, String path, List<ACL> acls) throws RuntimeException {
+
+        String npath = Utils.normalizePath(path);
+        if (npath.equals("/")) {
+            return;
+        }
+        if (existsNode(zk, npath, false)) {
+            return;
+        }
+        byte[] byteArray = new byte[1];
+        byteArray[0] = (byte) 7;
+        createNode(zk, npath, byteArray, org.apache.zookeeper.CreateMode.PERSISTENT, acls);
+
+    }
+
+    public static void syncPath(CuratorFramework zk, String path) throws RuntimeException {
+        try {
+            zk.sync().forPath(Utils.normalizePath(path));
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    public static void addListener(CuratorFramework zk, ConnectionStateListener listener) {
+        zk.getConnectionStateListenable().addListener(listener);
+    }
+
+    public static byte[] getData(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+
+        try {
+            String npath = Utils.normalizePath(path);
+            if (existsNode(zk, npath, watch)) {
+                if (watch) {
+                    return zk.getData().watched().forPath(npath);
+                } else {
+                    return zk.getData().forPath(npath);
+                }
+            }
+        } catch (KeeperException e) {
+            // this is fine b/c we still have a watch from the successful exists call
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+
+        return null;
+    }
+
+    public static Integer getVersion(CuratorFramework zk, String path, boolean watch) throws Exception {
```

Comment:
Could we please be explicit in the types of exception thrown, and not just Exception?